library/core/src/slice/ascii.rs

```diff
@@ -292,7 +293,16 @@ fn is_ascii(s: &[u8]) -> bool {
     // We also do this for architectures where `size_of::<usize>()` isn't
     // sufficient alignment for `usize`, because it's a weird edge case.
     if len < USIZE_SIZE || len < align_offset || USIZE_SIZE < mem::align_of::<usize>() {
-        return s.iter().all(|b| b.is_ascii());
+        // FIXME: once iterators and closures can be used in `const fn`,
+        // return s.iter().all(|b| b.is_ascii());
+        let mut i = 0;
+        while i < len {
```

Comment:
Instead of having to maintain an index, you could also use slice patterns to fake iteration:
```rust
while let [next, rest @ ..] = slice {
  slice = rest;
}
```